### ISSUE TITLE
MSBuildWorkspace.Create().OpenProjectAsync throws exception while loading .shproj type.

### ISSUE THREAD

**Issue by abhijitparkhi1983** [dotnet/roslyn#29696]
**Version Used**: 
2.8.0

**Steps to Reproduce**:

1.  We have multiple .csproj and .shproj projects in our application. 
2. We are using Roslyn for loading these projects and browse through each .cs documents from these .csproj and .shproj files using following code : 

```
public bool OpenProject(string projectPath)
{
                 using (var workspace = MSBuildWorkspace.Create())
                {
                    Project currentProject = workspace.OpenProjectAsync(projectPath).Result;
                    .....
                    ....
                }
}
```
3. The above code works well for all the .csproj files but when we try to run the same code for .shproj files, we get exception.

**Expected Behavior**:
The .shproj projects should load the way Roslyn loads .csproj projects.

**Actual Behavior**:
InvalidOperationException: Cannot open project '.../react-native-windows-master/ReactWindows\ReactNative.Shared\ReactNative.Shared.shproj' because the file extension '.shproj' is not associated with a language.

**Comment by jinujoseph** [comment]
cc @DustinCampbell

**Comment by abhijitparkhi1983** [comment]
@DustinCampbell @JieCarolHu  any update on this one please.

**Comment by Mobrockers** [comment]
@DustinCampbell Hey! We need this because consumers of our analyzer tool use these shared projects, can you give any guidance on how to do this or if this is at all possible with msbuildworkspace?

**Comment by AArnott** [comment]
.shproj projects *do not build*. So they needn't load like .csproj projects load. If Roslyn simply *skipped* these projects completely, that should unblock any read-only customer scenario.

**Comment by jmarolf** [comment]
My understanding is that they are skipped today. You still get a workspace it just does not contain any `shproj` projects

**Comment by Mobrockers** [comment]
@jmarolf If they are not included in the workspace then we cannot get the syntaxes for the source files included in the projects. Or am I missing something.

**Comment by AArnott** [comment]
Yes you can. The Compile items for them are in the other csproj files by virtue of an Import of a projitems file.

**Comment by jods4** [comment]
2022 update: `MSBuildWorkspace` has `SkipUnrecognizedProjects` (defaults to `true`) to decide if it should throw or log a diagnostic when encountering unsupported projects.

It's nice, my only wish is that those projects were logged as `Warning` diagnostics and not `Failure`.

I wrote a tool to process C# code and it logs in bright red on the console every `Failure`.
If the solution contains a `.esproj` for example, I don't care but it's a big red line in the face of users anyway.

The only solution I see would be to special case this message, i.e. sniff the message format, which is something that will fail badly on localized environments or possibly after framework upgrades.
